Source/OpenDRIVE/Private/OpenDRIVEActor.cpp

```diff
@@ -48,19 +68,29 @@ void AOpenDRIVEActor::LoadOpenDrive()
 	}
 }
 
+bool AOpenDRIVEActor::CheckForMultipleOpenDRIVEActors()
+{
+	TArray<AActor*> openDRIVEActorInScene;
+
+	UGameplayStatics::GetAllActorsOfClass(GetWorld(), AOpenDRIVEActor::StaticClass(), openDRIVEActorInScene);
+
+	return (openDRIVEActorInScene.Num() > 1);
+}
+
+bool AOpenDRIVEActor::CheckForMultipleOpenDRIVEActors(TArray<AActor*> &actors)
+{
+	UGameplayStatics::GetAllActorsOfClass(GetWorld(), AOpenDRIVEActor::StaticClass(), actors);
+
+	return (actors.Num() > 1);
+}
+
 #if WITH_EDITOR
 
 void AOpenDRIVEActor::PostEditMove(bool bFinished)
 {
-	TArray<AActor*> openDRIVEActorInScene;
-	
-	UGameplayStatics::GetAllActorsOfClass(GetWorld(), AOpenDRIVEActor::StaticClass(), openDRIVEActorInScene);
+	CheckForMultipleOpenDRIVEActors() ? GEngine->AddOnScreenDebugMessage(0, 10.f, FColor::Red, TEXT("Multiple OpenDRIVEActor detected. Must be only OpenDRIVEActor per scene. \n The plugin can't work correctly."))
```

Comment:
If you end up using the subsystem mentioned above the `PostEditMove` override will probably go away altogether, but just in case: I would advise against the current implementation. This means that `UGameplayStatics::GetAllActorsOfClass` (which can be rather slow in large scene) is called each time the actor is moved, which is... bad.

It's better to only deal with warnings when the issue arise (actor created, map with issue opened, etc.) or is solved (actor removed, new map opened, etc.). The rest of the time, we don't really care what happens.